### PR TITLE
[HGI-85] - Expose transaction variables to action destinations

### PR DESCRIPTION
**Pull request by varadarajan-tw**
JIRA Ticket - https://segment.atlassian.net/browse/HGI-85
This PR is to add support for Transaction variables. Hubspot Actions, like its classic version, has a dependency on properties passed in transaction variable.  Hubspot uses transaction variables to store and resolve hubspot id with segment user id. Pls refer [Notes](https://docs.google.com/document/d/1dfUxurwhUSR0W4JbeHSGJYiz5u574uXdCquBG-cdyeA/edit)

Integrations PR - https://github.com/segmentio/integrations/pull/2385

## Testing

_Include any additional information about the testing you have completed to
ensure your changes behave as expected. For a speedy review, please check
any of the tasks you completed below during your testing._

- [x] Added [unit tests](https://github.com/segmentio/action-destinations/blob/main/docs/testing.md#local-end-to-end-testing) for new functionality
- [x] Tested end-to-end using the [local server](https://github.com/segmentio/action-destinations/blob/main/docs/testing.md#local-end-to-end-testing)
- [x] [Segmenters] Tested in the staging environment

### **Staging test results**

**Workspace Link**
https://app.segment.build/varadarajan-testing/destinations/actions-hubspot-cloud/sources/http_api2/instances/632d970f5197352dd701f249/actions/eHKh6TwagDR9Maqh7DG8Mp

- Deployed HubSpot action destination for verifying if the transaction variables are received and set properly. Added Log Statements to validate
[Log statement in Contact Action](https://github.com/segmentio/action-destinations/blob/tx-variables/packages/destination-actions/src/destinations/hubspot/contact/index.ts#L70-L71)
[Log statement in Company Action](https://github.com/segmentio/action-destinations/blob/tx-variables/packages/destination-actions/src/destinations/hubspot/company/index.ts#L56-L58)
[Log statement in Custom Behavioral Action](https://github.com/segmentio/action-destinations/blob/f339c2f869b2cf037982f56bae4131395be7749b/packages/destination-actions/src/destinations/hubspot/sendCustomBehavioralEvent/index.ts#L66-L67)
- Posted an `Identify` event to trigger Contact Action and `Group` event to trigger Company Action. Log results below
image
- Posted a `track` event to trigger `Custom Behavioral Event` that doesn't involve identity GX. No transaction variables are received as expected.
image

### CODE REVIEW

File: packages/core/src/destination-kit/index.ts [hunk 1]
```diff
@@ -195,6 +197,12 @@ interface OnEventOptions {
   features?: Features
   statsContext?: StatsContext
   logger?: Logger
+  transactionContext?: TransactionContext
+}
+
+export interface TransactionContext {
```
Comment (dlasky) [inline packages/core/src/destination-kit/index.ts:203]: can you add a JSDOC here about this being private as well

**Review by dlasky — APPROVED** [review]

File: packages/core/src/destination-kit/index.ts (hunk 1 above)
Comment (varadarajan-tw) [reply packages/core/src/destination-kit/index.ts:203]: Sure! Adding JSDOC as suggested.

File: packages/core/src/__tests__/destination-kit.test.ts [hunk 2]
```diff
@@ -427,4 +434,47 @@ describe('destination kit', () => {
...
+          }
+        }
+      }
+      const eventOptions = {
+        features: {},
+        statsContext: {} as StatsContext,
+        logger: { name: 'test-integration', level: 'debug' } as Logger,
+        transactionContext: {
+          transaction: { contact_id: '801' },
+          setTransaction: (key: string, value: string) => console.log(key, value)
```
Comment (brennan) [inline packages/core/src/__tests__/destination-kit.test.ts:463]: I know this is a unit test, but should we remove the `console.log` here?

Comment (varadarajan-tw) [reply packages/core/src/__tests__/destination-kit.test.ts:463]: Sure. I will have this incorporated in next PR that raise for hubspot actions if that is okay!

Comment (brennan) [reply packages/core/src/__tests__/destination-kit.test.ts:463]: Yes, sounds good! Thanks!

**Pull request merged**